### DISCUSSION TITLE
Using Github action integrating Crowdin to translate the website

### DISCUSSION THREAD

**Discussion by ErikQQY** [facebook/docusaurus#4911] [Q&A] [answered]
Hello

I am using Github action with Crowdin to translate the website.

When I set the command ```"crowdin:sync" : "docusaurus write-translations && crowdin upload && crowdin download"```, and Github action excutes the command automatically, the Build and Deploy of my website is successful, however, there are still some issues.

As for French and German translations, it's working, and the github action log is OK:

[![2tBL4O.png](https://z3.ax1x.com/2021/06/05/2tBL4O.png)](https://imgtu.com/i/2tBL4O)

But the Chinese translation and Korean translation is not working, and the Github action log shows this:

[![2tDiUf.png](https://z3.ax1x.com/2021/06/05/2tDiUf.png)](https://imgtu.com/i/2tDiUf)

![ko](https://user-images.githubusercontent.com/52615090/120881790-323e9e00-c606-11eb-945d-09ef0ba12f02.png)

I think it is an error relating to Blogs, but the Blogs this website using is the default blogs in Docusaurus.

The website is https://casdoor.org

Any answers is appreciated.

**Comment by slorber** [comment, marked as answer]
> I am using Github action with Crowdin to translate the website.

Please make sure that you are able to build your localized site locally before using any CI tool. That only complicates the debugging.

> But the Chinese translation and Korean translation is not working

I don't know what "is not working" means for you. Have you translated content in Crowdin, and this translation does not appear after building your website in some languages? Have you verified that the downloaded crowdin translations are at the correct location? 

> and the Github action log shows this:

This error log is not a blocker issue:

![image](https://user-images.githubusercontent.com/749374/121065873-8645b000-c7c9-11eb-8646-66b97e88223b.png)

Each language has its own plural rules and chinese/korean only have one. The translated string should only contain one plural form like `{count} min read` instead of 2.  (ie the message should not contain any `|` separator char)

This warning should not prevent your site from displaying your chinese/korean markdown content.

Normally those strings are from the classic template, and we provide default translations for those elements, but when using Crowdin somehow you have to retranslate those strings using the Crowdin UI. I'd like to improve the DX regarding this problem, not  sure yet how, as it's a bit annoying to have to provide translations for strings that are already translated by default

**Reply by ErikQQY** [reply]
Thank you for your detailed reply!!!!

I think it is the ```locales: ['en', 'zh-ch', 'fr']``` in ```docusaurus.config.js``` that makes the translation on Crowdin doesn't work. After I changed it to ```locales: ['en', 'zh', 'fr'], it works perfectly.

Thank you again for your helping.😄
